src/Pos/Util.hs

```diff
@@ -0,0 +1,377 @@
+import qualified Control.Monad                    as Monad (fail)
+import           Control.Monad.STM                (retry)
+import           Control.Monad.Trans.Resource     (ResourceT)
+import           Data.Binary                      (Binary)
+import qualified Data.Cache.LRU                   as LRU
+import           Data.Hashable                    (Hashable)
+import qualified Data.HashMap.Strict              as HM
+import           Data.List                        (span, zipWith3)
+import qualified Data.List.NonEmpty               as NE
+import           Data.SafeCopy                    (SafeCopy (..), base, contain,
+                                                   deriveSafeCopySimple, safeGet, safePut)
+import qualified Data.Text                        as T
+import qualified GHC.Exts                         as IL
+import qualified Language.Haskell.TH              as TH
+import           Mockable                         (Mockable, Throw, throw)
+import           Serokell.Util                    (VerificationRes (..))
+import           System.Wlog                      (LoggerNameBox (..))
+import           Test.QuickCheck                  (Arbitrary)
+import           Text.Parsec                      (ParsecT)
+import           Unsafe                           (unsafeInit, unsafeLast)
+-- SafeCopy instance for HashMap
+import           Serokell.AcidState               ()
+
+import           Pos.Binary.Class                 (Bi)
+import           Pos.Util.Arbitrary
+import           Pos.Util.TimeLimit
+import           Pos.Util.Undefined               ()
+import           Pos.Util.Util
+
+mappendPair :: (Monoid a, Monoid b) => (a, b) -> (a, b) -> (a, b)
+mappendPair = (uncurry (***)) . (mappend *** mappend)
+
+mconcatPair :: (Monoid a, Monoid b) => [(a, b)] -> (a, b)
+mconcatPair = foldr mappendPair (mempty, mempty)
+
+-- | Concatenates two url part using regular slash '/'.
+-- E.g. @"./dir/" <//> "/file" = "./dir/file"@.
+(<//>) :: String -> String -> String
+(<//>) lhs rhs = lhs' ++ "/" ++ rhs'
+  where
+    isSlash = (== '/')
+    lhs' = reverse $ dropWhile isSlash $ reverse lhs
+    rhs' = dropWhile isSlash rhs
+
+-- | Convert (Reader s) to any (MonadState s)
+readerToState
+    :: MonadState s m
+    => Reader s a -> m a
+readerToState = gets . runReader
+
+deriveSafeCopySimple 0 'base ''VerificationRes
+
+-- | A helper for simple error handling in executables
+eitherPanic :: Show a => Text -> Either a b -> b
+eitherPanic msgPrefix = either (error . (msgPrefix <>) . show) identity
+
+-- | This function performs checks at compile-time for different actions.
+-- May slowdown implementation. To disable such checks (especially in benchmarks)
+-- one should compile with: @stack build --flag cardano-sl:-asserts@
+inAssertMode :: Applicative m => m a -> m ()
+#ifdef ASSERTS_ON
+inAssertMode x = x *> pure ()
+#else
+inAssertMode _ = pure ()
+#endif
+{-# INLINE inAssertMode #-}
+
+-- | Remove elements which are in 'b' from 'a'
```

Comment:
Seems to be an unused function. And doesn't seem correct:

```
> diffDoubleMap (HM.fromList [(1,(HM.fromList []))]) (HM.fromList [(1,(HM.fromList [(1,())]))])
fromList []
```

There's also [a test suite for this](https://github.com/fpco/cardano-sl/blob/72db714533cc4ef8d9f4df916d55fdea5b7745f5/test/Test/Pos/UtilSpec.hs#L1). 

I suggest removing the function and test suite.